zipline/finance/commission.py

```diff
@@ -12,113 +12,162 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import abc
+
+from abc import abstractmethod
+from six import with_metaclass
 
 DEFAULT_PER_SHARE_COST = 0.0075         # 0.75 cents per share
 DEFAULT_MINIMUM_COST_PER_TRADE = 1.0    # $1 per trade
 
 
-class PerShare(object):
-    """Calculates a commission for a transaction based on a per
-    share cost with an optional minimum cost per trade.
+class CommissionModelBase(with_metaclass(abc.ABCMeta)):
+    """
+    Abstract commission model interface.
+    """
+
+    @abstractmethod
+    def calculate(self, order, transaction):
+        """
+        Parameters
+        ----------
+        order: the order whose transaction we are processing.  Its `commission`
+            field is up to date with how much commission has been attributed
+            to this order so far.
+
+        transaction: the transaction we are processing.
+
+        Returns
+        -------
+        float: The additional commission we should attribute to this order.
+        """
+        pass
+
+
+class PerShare(CommissionModelBase):
+    """
+    Calculates a commission for a transaction based on a per share cost with
+    an optional minimum cost per trade.
 
     Parameters
     ----------
     cost : float, optional
         The amount of commissions paid per share traded.
     min_trade_cost : optional
-        The minimum amount of commisions paid per trade.
+        The minimum amount of commissions paid per trade.
     """
 
     def __init__(self,
                  cost=DEFAULT_PER_SHARE_COST,
                  min_trade_cost=DEFAULT_MINIMUM_COST_PER_TRADE):
-        """
-        Cost parameter is the cost of a trade per-share. $0.03
-        means three cents per share, which is a very conservative
-        (quite high) for per share costs.
-        min_trade_cost parameter is the minimum trade cost
-        regardless of the number of shares traded (e.g. $1.00).
-        """
-        self.cost = float(cost)
-        self.min_trade_cost = None if min_trade_cost is None\
-            else float(min_trade_cost)
+        self.cost_per_share = float(cost)
+        self.min_trade_cost = None
```

Comment:
Why not just `self.min_trade_cost = min_trade_cost`.  Not sure what the use-case is for passing a non-float.  (This also means we're actually accepting things like strings here...)